pkg/plugin/plugin.go

```diff
@@ -100,6 +109,42 @@ func (p *Plugin) SetInformation(parsedURL *url.URL) error {
 	return nil
 }
 
+func findPluginByName(name string, plugins []store.Plugin) *store.Plugin {
+	// for each plugin in the plugins list, check if the name matches the name of the plugin
+	for _, plugin := range plugins {
+		if plugin.Name == name {
+			return &plugin
+		}
+	}
+
+	return nil
+}
+
+func (p *Plugin) checkForPluginUpdates() (bool, error) {
+	// finf if there is an element elem in storeitems that has the same name as rootItem.Name()
+	storeItems, err := store.FetchPluginList()
+	if err != nil {
+		return false, fmt.Errorf("while fetching store list: %w", err)
+	}
+
+	pluginVersion, err := version.NewVersion(p.info.Version)
+	if err != nil {
+		return false, fmt.Errorf("while parsing plugin version: %w", err)
+	}
+
+	elemInStore := findPluginByName(p.info.Name, storeItems)
```

Comment:
Why not naming it plugin instead of elemInStore ?

Reply:
because its type is store.Plugin and not Plugin, 
indeed I could make the naming clearer